### ISSUE TITLE
bug(material/tooltip): tooltip inside sidenav hides after text update

### ISSUE THREAD

**Issue by amanosacrous** [angular/components#27782]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When the tooltip component overflows a parent sidenav and its text content is updated dynamically, it hides.

We have debugged the code, and this happens because the sidenav component extends the cdk-scrollable.

The piece of code that calls the tooltip hide function is the following:
https://github.com/angular/components/blob/bae953999407fd385b1d571e04350bf0f28abf05/src/material/tooltip/tooltip.ts#L510-L520

### Reproduction

Stackblitz url:
https://stackblitz.com/edit/stackblitz-starters-wbd1hh?file=src%2Ftooltip-sidenav%2Ftooltip-sidenav.component.html

As you can see on the stackBlitz example, we have two tooltips:
- The tooltip inside the sidenav, which overflows the sidenav component. That if we update the tooltip text, the tooltip disappears. (BUG)
- On the other hand, we have the tooltip inside the mat-sidenav-content which works properly, and it doesn't hide after updating it. (WORKS WELL because doesn't overflow the parent)

### Expected Behavior

The tooltip inside the sidenav should not hide when the text is updated.

### Actual Behavior

The tooltip placed inside the sidnav that overflows is hidden suddenly when the text is updated.

### Environment

- Angular: v16.2.4
- CDK/Material: v16.2.3
- Browser(s): Google chrome Versión 116.0.5845.187, Firefox 117.0
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by amysorto** [comment]
Having a tooltip within a sidenav doesn't seem like a common use case for most apps, it's unlikely this will be priortized in the near future. This issue can be reopened if that changes later.

**Comment by llorenspujol** [comment]
I don't believe we should close this issue so quickly, @amysorto. We put a lot of effort into providing this report. It includes a detailed description of the issue, a StackBlitz reproduction, and a link to the code that is causing the tooltip to hide.

If it won't be fixed, perhaps we could label it as 'wontfix' or something similar, rather than closing it. Closing the issue might give the impression that it's resolved, which it isn't. Providing a workaround would be enough. I'm sure that one of our core members familiar with the overlay/tooltip codebase can provide one.

Keeping the repository with less issues is good, I agree, but in this case it does not feel right. Tagging the issue with both 'bug' and 'wontfix' would be better IMO.

**Comment by andrewseguin** [comment]
Thank you for the follow up and for sharing your thoughts! We appreciate the work you put in the issue and the reproduction. As we mentioned above, this is not a common use case, but we’ll reopen and add the “help wanted” label in case anyone from the community has the bandwidth to fix it.

**Comment by SoaringColinLaws** [comment]
Funny enough, we do use matToolTip in our nav as well, and I found this issue while trying to figure out why tool tips aren't working at all for us in Ng 15. We had some scss to temporarily fix the issue that existed in Ng 14, but it appears that tooltips are broken again. For what it's worth, I figured to just mention that yes, we also use tooltips in our navigation.

**Comment by czhang0405** [comment]
After 2 hour debug, I found the problem. Most of all, let's dive into it.
- Before I start explain, Let's see when  `scrollableViewProperties.isOverlayClipped` is `true`, follow spec: 

https://github.com/angular/components/blob/7cb6f3152d627a2e63636da8ed6d405a5bc92092/src/cdk/overlay/position/connected-position.ts#L59-L76

Steps: 

1. Everything start from this function

https://github.com/angular/components/blob/bae953999407fd385b1d571e04350bf0f28abf05/src/material/tooltip/tooltip.ts#L668

2. After that, it will come to this block and get new value for `scrollableViewProperties` is `true`.

https://github.com/angular/components/blob/df60e733c60e572ba538f6ad0ceff3e63e527b53/src/cdk/overlay/position/flexible-connected-position-strategy.ts#L719-L723

![Untitled](https://github.com/angular/components/assets/42007633/f60927f4-9153-4e01-adb3-9b9693abcbf2)

3. Finally, it will emit `positionChanges` event and hide tooltip

https://github.com/angular/components/blob/bae953999407fd385b1d571e04350bf0f28abf05/src/material/tooltip/tooltip.ts#L510-L520

Solutions:

- Solution 1: you need to make Scrollable content larger than tooltip.

https://stackblitz.com/edit/stackblitz-starters-7zrzcu?file=src%2Ftooltip-sidenav%2Ftooltip-sidenav.component.ts,src%2Ftooltip-sidenav%2Ftooltip-sidenav.component.html,src%2Ftooltip-sidenav%2Ftooltip-sidenav.component.css

- Solution 2: use `runOutsideAngular`

https://stackblitz.com/edit/stackblitz-starters-zqpb3p?file=src%2Ftooltip-sidenav%2Ftooltip-sidenav.component.ts

**Comment by czhang0405** [comment]
Hi @amysorto , would it be possible to add a property named `autoHideOnClip` of type `boolean` to determine whether the tooltip can be hidden when it is clipped ?

**Comment by abdul03829** [comment]
Hi team,

I noticed that a solution for this issue was provided on July 27,by @czhang0405. It seems like a solid solution and would be great to see it merged.

Is there anything else needed to move this forward? Let me know if I can assist in any way.

Thanks,
Abd-ur-Rahman
